packages/nvidia_nat_mcp/src/nat/plugins/mcp/client_base.py

```
@@ -599,25 +626,18 @@ async def acall(self, tool_args: dict) -> str:
-            # on first tool call
-            auth_callback = _Ctx.get().user_auth_callback
-            if auth_callback and self._parent_client:
-                # set custom auth callback
-                self._parent_client.set_user_auth_callback(auth_callback)
-
-            # get session id from context, authentication is done per-websocket session for tool calls
-            cookies = getattr(_Ctx.get().metadata, "cookies", None)
-            if cookies:
-                session_id = cookies.get("nat-session")
+            session_id = self._get_session_id()
         except Exception:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)